### PR TITLE
* Update doc with developer section for creating own models

### PR DESCRIPTION
**Pull request by DocGarbanzo**
* Adding new page with detailed info how to create own models
* Update donkey command section with `donkey train` command.

### CODE REVIEW

File: docs/dev_guide/model.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+```
+
+In this function you define how to extract the input data from your
+recorded data. This data is usually called `X` in the ML frame work . We have
+shown the implementation in the base class which works for all models that have
+only the image as input. 
+
+The function returns a single data item if the model has only one input. You 
+need to return a tuple if your model uses more input data.
+**Note:** _If your model has more inputs, the tuple needs to have the image in
```
Comment (tikurahul) [inline docs/dev_guide/model.md:70]: Add 2 extra lines to emphasize the `Note`.  (Creates a new paragraph)

File: docs/dev_guide/model.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    return {'img_in': x}
+```
+Here we require a translation of how the `X` value that you extracted above will
+be fed into `tf.data`. Note, `tf.data` expects a dictionary if the model has
+more than one input variable, so we have chosen to use dictionaries also in the
+one-argument case for consistency. Above we have shown the implementation in the
+base class which works for all models that have only the image as input. You
+don't have to overwrite neither `x_transform` nor
+`x_translate` if your model only uses the image as input data.
+**Note:** _the keys of the dictionary must match the name of the **input**
```
Comment (tikurahul) [inline docs/dev_guide/model.md:94]: Same as above.

File: docs/dev_guide/model.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    img_shape = self.get_input_shape()[1:]
+    shapes = ({'img_in': tf.TensorShape(img_shape)},
+              {'angle_out': tf.TensorShape([15]),
+               'throttle_out': tf.TensorShape([20])})
+    return shapes
+```
+This function returns a tuple of _two_ dictionaries that tells tensorflow which
+shapes are used in the model. We have shown the example of the 
+`KerasCategorical` model here.
+**Note 1:** _The keys of the two dictionaries must match the name of the
```
Comment (tikurahul) [inline docs/dev_guide/model.md:123]: Nit: Call this **Note** and not **Note 1**. 
Add a paragraph like the ones above.

File: docs/dev_guide/model.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    outputs = self.model.predict(img_arr)
+    steering = outputs[0]
+    throttle = outputs[1]
+    return steering[0][0], throttle[0][0]
+```
+Here we are showing the implementation of the linear model. Please note that 
+the input tensor shape always contains the batch dimension in the first 
+place, hence the shape of the input image is adjusted from 
+`(120, 160, 3) -> (1, 120, 160, 3)`.
+**Note:** _If you are passing another array in the`other_arr` variable, you will
```
Comment (tikurahul) [inline docs/dev_guide/model.md:149]: Same here.

**Review by tikurahul — APPROVED** [review]

File: docs/dev_guide/model.md (hunk 1 above)
Comment (DocGarbanzo) [reply docs/dev_guide/model.md:70]: yup

File: docs/dev_guide/model.md (hunk 2 above)
Comment (DocGarbanzo) [reply docs/dev_guide/model.md:94]: yup

File: docs/dev_guide/model.md (hunk 3 above)
Comment (DocGarbanzo) [reply docs/dev_guide/model.md:123]: I have 2 notes there...

**Pull request merged**